handshake_client.go

```diff
@@ -411,6 +414,48 @@ func (hs *clientHandshakeState) processCertsFromServer(certificates [][]byte) er
 	return nil
 }
 
+// processDelegatedCredentialFromServer unmarshals the DelegatedCredential
+// offered by the server (if present) and validates it using the peer
+// certificate.
+func (hs *clientHandshakeState) processDelegatedCredentialFromServer(dc []byte) error {
+	c := hs.c
+
+	var cred *DelegatedCredential
+	var err error
+	if dc != nil {
+		if !hs.hello.delegatedCredential {
+			c.sendAlert(alertUnexpectedMessage)
+			return errors.New("tls: got delegated_credential extension without indication")
+		}
+
+		cred, err = unmarshalDelegatedCredential(dc)
+		if err != nil {
+			c.sendAlert(alertDecodeError)
+			return fmt.Errorf("tls: delegated_credential: %s", err)
+		}
+	}
+
+	if cred != nil && !c.config.InsecureSkipVerify {
+		var now time.Time
+		if hs.c.config.Time != nil {
+			now = hs.c.config.Time()
+		} else {
+			now = time.Now()
+		}
+
+		if v, err := cred.Validate(c.peerCertificates[0], hs.c.vers, now); err != nil {
```

Comment:
``var err`` is already above, no need for ``:=``

Reply:
But `v` is undefined, hence the need for `:=`.

Reply:
okis

Reply:
Donel.